### ISSUE TITLE
Support `CHECK` constraints in MariaDB

### ISSUE THREAD

**Issue by CodingCellist** [dbeaver/dbeaver#6069]
When looking at, or creating, constraints of tables in a MariaDB database, `CHECK` constraints don't show up, nor can they be created. Previously in MariaDB, `CHECK` was ignored, but as of MariaDB 10.2.1 `CHECK` constraints [are no longer ignored](https://mariadb.com/kb/en/library/constraint/#check-constraints).

Tables using `CHECK` constraints can be created by taking the DDL of the table, pasting it into a SQL-editor, manually specifying the constraints, and then running the script. However, there is no indication in the "Constraints" tab that these constraints are there.

Being able to use these constraints directly in the "Create new table" option, and having them show up if/when they are there, would be very useful.

**Comment by darielvicedo** [comment]
CHECK constraints are needed in MariaDB for the JSON data type. I need this feature in the 'Create new table' view.

**Comment by LonwoLonwo** [comment]
Ability to create `CHECK` constraints added for MariaDB version at least 10.2 and MySQL version at least 8.0

**Comment by kseniiaguzeeva** [comment]
MySQL 8 doesn't support CHECK constraints (build 7.1.1.202006231441).
Other points are not tested yet

**Comment by kseniiaguzeeva** [comment]
Tested on MariaDB 10.3:

1 Deleting and creating Check constraint via SQL Editor is visible only after reconnecting to the database

2 If Check constraint is created via SQL Editor and has the column name, you cannot delete it (both UI and SQL Editor). Perhaps the reason is in quoting the name check constraint in the condition.
Steps:
a. create a table
CREATE TABLE t2 (name VARCHAR (30) CHECK (CHAR_LENGTH (name)> 2), start_date DATE,
  end_date DATE CHECK (start_date IS NULL OR end_date IS NULL OR start_date <end_date));
b. open the Check constraint tab
c. try to delete a Check constraint
Actual result: an error appears, the object is not deleted.

3 Check constraint is not expanded if it is created via UI.
Steps:
a. Create table CREATE TABLE t1 (a INT, b INT)
b. Create for the table Check constraint via UI:
ALTER TABLE t1 ADD CONSTRAINT a CHECK (a>2);
ALTER TABLE t1 ADD CONSTRAINT b CHECK (b>2);
c. Open Check constraints tab in the table Metadata Editor
d. press the arrow to expand the tree.
Actual result:
The arrow dissapears, now new objects under it.
![lJp37v0dux](https://user-images.githubusercontent.com/51405061/85742100-a476e500-b70b-11ea-93c7-530da4e2501c.gif)

4 Check expression field in Check constraints properties truncates the long Check constraint expression
Steps:
a. Open tab check contraints for t2
b. Expand column check expression for end_date
![4](https://user-images.githubusercontent.com/51405061/85739946-e4d56380-b709-11ea-9f8d-6a07c7a796ce.png)

**Comment by LonwoLonwo** [comment]
1. - fixed
3. - fixed. check constraint is not expanding now

2. this problem occurs only in MariaDB, MySQL is fine. 
Workaround. Do not name the check constraint by the column name. Give the check constraint names, while creating via SQL Editor.
4. unfortunately, check_clause column in information_shema.check_constraints is limited. We will try to solve this problem in the future
![2020-06-26 12_15_55-DBeaver 7 1 1 - _localhost_ Script-27](https://user-images.githubusercontent.com/45152336/85985526-84893f00-b9f3-11ea-8859-9a924c757a72.png)

**Comment by kseniiaguzeeva** [comment]
1 verified
2 verified
3 do not have an enviroment to test. Check constraint custom names can be assigned also via UI.
4 Ok, new ticket will be created.

**Comment by ndiaz-addepar** [comment]
> MySQL 8 doesn't support CHECK constraints (build 7.1.1.202006231441).
> Other points are not tested yet

This has not been true since 8.0.16
